### PR TITLE
Embed spartan_wasm.wasm & Remove circuit artifacts from lib

### PR DESCRIPTION
**Pull request by DanTehrani**
- Embed spartan_wasm bytes into a typescript file using [embed_wasm_bytes.ts](https://github.com/personaelabs/spartan-ecdsa/blob/dan/embed-spartan-wasm/packages/lib/embed_wasm_bytes.ts) at build time.
- Removed circuit artifacts from the lib to make the package small enough.
    - (Previously, if the lib was used in node.js then the user could simply pass in the default config to use the circuit artifacts packaged inside the lib. With this update, even if the lib is used in a node.js env, the lib user must download the circuit and wasm files and specify the path to them when instantiating prover/verifier.
    - This results in
        - A simpler build process
        - A smaller package (w/ the circuit artifacts included, the package resulted in > 100MB and without it it’s 2MB)
- Log warning message when default config is used
    - [https://github.com/personaelabs/spartan-ecdsa/blob/be0df0a8a3b8571b287c22f58451659026039c73/packages/lib/src/core/membership_prover.ts#L36](https://github.com/personaelabs/spartan-ecdsa/blob/be0df0a8a3b8571b287c22f58451659026039c73/packages/lib/src/core/membership_prover.ts#L36)
    - [https://github.com/personaelabs/spartan-ecdsa/blob/be0df0a8a3b8571b287c22f58451659026039c73/packages/lib/src/core/membership_verifier.ts#L23](https://github.com/personaelabs/spartan-ecdsa/blob/be0df0a8a3b8571b287c22f58451659026039c73/packages/lib/src/core/membership_verifier.ts#L23)

### CODE REVIEW

**Comment by DanTehrani** [comment]
@lsankar4033 
Could u review the changes I wrote about [here](https://github.com/personaelabs/spartan-ecdsa/pull/20#issue-1564085984)?  
Easier/better designs and suggestions are welcome!

File: packages/lib/src/core/membership_prover.ts [hunk 1]
```diff
@@ -11,27 +10,43 @@ import {
...
 
-    this.leafType = options.leafType;
+    if (
+      options.circuit === defaultPubkeyMembershipPConfig.circuit ||
+      options.witnessGenWasm ===
+        defaultPubkeyMembershipPConfig.witnessGenWasm ||
+      options.circuit === defaultAddressMembershipPConfig.circuit ||
+      options.witnessGenWasm === defaultAddressMembershipPConfig.witnessGenWasm
+    ) {
+      console.warn(`
```
Comment (lsankar4033) [inline packages/lib/src/core/membership_prover.ts:36]: excellent

File: packages/lib/src/core/membership_verifier.ts [hunk 2]
```diff
@@ -27,11 +41,12 @@ export class MembershipVerifier extends Profiler implements IVerifier {
...
-    const result = await this.spartanWasm.verify(
-      circuitBin,
-      proof,
-      publicInput
-    );
+    let result;
+    try {
+      result = await wasm.verify(circuitBin, proof, publicInput);
+    } catch (_e) {
+      result = false;
```
Comment (lsankar4033) [inline packages/lib/src/core/membership_verifier.ts:48]: good catch...

File: packages/lib/src/wasm/wasm.js [hunk 3]
```diff
@@ -400,16 +400,19 @@ function finalizeInit(instance, module) {
     return wasm;
 }
 
-function initSync(module, maybe_memory) {
+async function initSync(module, maybe_memory) {
```
Comment (lsankar4033) [inline packages/lib/src/wasm/wasm.js:403]: is this method idempotent? can't quite tell. if not, should we have logic in the calling function not to double initialize?

**Review by lsankar4033 — CHANGES_REQUESTED** [review]
overall lgtm pending the one question abt double-initialization!

**Comment by lsankar4033** [comment]
oh also just a note that the failing membership tests pass, but console.error in test logs:
```
console.error
  panicked at 'called `Result::unwrap()` on an `Err` value: Custom("elliptic curve point encoding error")', packages/spartan_wasm/
wasm.rs:60:23
  
  Stack:
  
  Error: 
      at imports.wbg.__wbg_new_abda76e883ba8a5f (/Users/lakshmansankar/src/spartan-ecdsa/packages/lib/src/wasm/wasm.js:361:21)
      at null.<anonymous> (wasm://wasm/001e7302:1:387360)
      at null.<anonymous> (wasm://wasm/001e7302:1:445282)
      at null.<anonymous> (wasm://wasm/001e7302:1:417454)
      at null.<anonymous> (wasm://wasm/001e7302:1:431450)
      at null.<anonymous> (wasm://wasm/001e7302:1:442714)
      at null.<anonymous> (wasm://wasm/001e7302:1:438764)
      at null.<anonymous> (wasm://wasm/001e7302:1:441875)
      at null.<anonymous> (wasm://wasm/001e7302:1:433946)
      at null.<anonymous> (wasm://wasm/001e7302:1:289649)
      at null.<anonymous> (wasm://wasm/001e7302:1:431007)
      at Object.verify (/Users/lakshmansankar/src/spartan-ecdsa/packages/lib/src/wasm/wasm.js:122:14)
      at MembershipVerifier.<anonymous> (/Users/lakshmansankar/src/spartan-ecdsa/packages/lib/src/core/membership_verifier.ts:46:2

      at Generator.next (<anonymous>)
      at fulfilled (/Users/lakshmansankar/src/spartan-ecdsa/packages/lib/src/core/membership_verifier.ts:28:58)
  ```

Think this is fine for now as tests properly pass, but we'll probably need to suppress these when we set up CI

File: packages/lib/src/wasm/wasm.js (hunk 3 above)
Comment (DanTehrani) [reply packages/lib/src/wasm/wasm.js:403]: @lsankar4033  It is idempotent but yeah, better to initialize it only once for performance.
Modified here: 9ddf979a462313ccc83d37f5f406c1c65701c9ad

**Comment by DanTehrani** [comment]
@lsankar4033 
Good point on console.error. Logging can be controlled by calling/not calling `wasm.init_panic_hook()`.
(We can later disable it for CI)

**Review by lsankar4033 — APPROVED** [review]

**Comment by lsankar4033** [comment]
all lgtm!

**Pull request merged**